### ISSUE TITLE
Support for Go

### ISSUE THREAD

**Issue by mahmoudhossam** [andythigpen/nvim-coverage#5]
Go has built in support for test coverage through `go test -cover`

It'd be nice to see it supported as well.

**Comment by mahmoudhossam** [comment]
@andythigpen Thanks a lot!

I'll try it out and let you know if I run into any problems.

**Comment by matsp** [comment]
@mahmoudhossam Did you test this successfully? For me nvim-coverage is not loading the coverage.out file that will be produced by vim-test (after adding the custom optiona to the gotest command). I also activated auto_refresh but without luck. Did I miss something @andythigpen ?

**Comment by andythigpen** [comment]
@matsp If you can generate a small test case or a set of steps to reproduce your issue, then I can try to help out. Otherwise without more information, I'm not sure.

**Comment by obreitwi** [comment]
I had a similar issue: This was due to the fact that `nvim-coverage` assumes the `go.mod` to reside in the current working directory.

Once I pointed it to the correct file (in my case `../../go.mod` it worked flawlessly). I think the correct way to fix this would be to look for `go.mod` recursively up the file tree, but `plenary.nvim` does not yet seem to provide that functionality out-of-the-box, right?
